### PR TITLE
3174 API/Contact: Update

### PR DESCRIPTION
**Pull request by sorja**

### CODE REVIEW

File: src/client/store/data/actions/upsertContact.ts [hunk 1]
```diff
@@ -15,32 +13,31 @@ type Props = CycleDataParams & {
...
+const _update = async (props: Props): Promise<void> => {
+  const { assessmentName, cycleName, countryIso, sectionName, contact, field, raw } = props
+
+  const body = { contact, field, raw }
+  const params = { assessmentName, cycleName, countryIso, sectionName }
+  await axios.put(ApiEndPoint.CycleData.Contacts.one(), body, { params })
+}
+
+const create = Functions.debounce(_create, 1_000, 'createContact')
+const update = Functions.debounce(_update, 1_000, 'updateContact')
```
Comment (minotogna) [inline src/client/store/data/actions/upsertContact.ts]: better 5000 ?

Comment (minotogna) [reply src/client/store/data/actions/upsertContact.ts]: plus, the update throttle should be by field name as well

Comment (sorja) [reply src/client/store/data/actions/upsertContact.ts]: Ok, I update both

When I add the button to create, we will deprecate the create function in this file anyway

File: src/server/controller/cycleData/contact/update.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+import { BaseProtocol, DB } from 'server/db'
+import { NodeExtRepository } from 'server/repository/assessmentCycle/nodeExt'
+import { ActivityLogRepository } from 'server/repository/public/activityLog'
+
+type Props = {
+  assessment: Assessment
+  cycle: Cycle
+  countryIso: CountryIso
+  sectionName: string
```
Comment (minotogna) [inline src/server/controller/cycleData/contact/update.ts]: do we have SectionName type ?

File: src/server/api/cycleData/contacts/update.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+export const updateContact = async (req: CycleDataRequest<never, Body>, res: Response) => {
+  try {
+    const { assessmentName, cycleName, countryIso, sectionName } = req.query
+    const { contact, field, raw } = req.body
+
+    const { assessment, cycle } = await AssessmentController.getOneWithCycle({ assessmentName, cycleName })
+    const user = Requests.getUser(req)
+
+    const props = { assessment, cycle, countryIso, sectionName, user, nodeExt: contact[field], raw }
+    const createdContact = await CycleDataController.Contacts.update(props)
```
Comment (minotogna) [inline src/server/api/cycleData/contacts/update.ts]: this is more an updated contact

File: src/server/repository/assessmentCycle/nodeExt/upsert.ts [hunk 4]
```diff
@@ -23,6 +23,8 @@ export const upsert = <T extends NodeExt<unknown>>(props: Props, client: BasePro
          insert into ${schemaCycle}.node_ext 
            (country_iso, parent_uuid, props, type, uuid, value)
          values ($1, $2, $3::jsonb, $4, $5, $6::jsonb)
+             on conflict (uuid) do update
+              set (parent_uuid, props, value) = ($2, $3::jsonb, $6::jsonb)
```
Comment (minotogna) [inline src/server/repository/assessmentCycle/nodeExt/upsert.ts]: can we really change a parent uuid ? maybe not?

File: src/server/repository/assessmentCycle/countryActivityLog/_common/acceptedMessages.ts [hunk 5]
```diff
@@ -1,10 +1,13 @@
 import { ActivityLogMessage } from 'meta/assessment'
 
 export const acceptedMessages = [
+  ActivityLogMessage.assessmentStatusUpdate,
+  ActivityLogMessage.contactUpdate,
```
Comment (minotogna) [inline src/server/repository/assessmentCycle/countryActivityLog/_common/acceptedMessages.ts:6]: we might need contact create as well?

Comment (minotogna) [reply src/server/repository/assessmentCycle/countryActivityLog/_common/acceptedMessages.ts:6]: do we have them in the last edit country summary view as well ?

**Comment by minotogna** [comment]
@sorja why was 
Screenshot 2023-12-21 at 12 21 39

merged here?
you're forcing double review

**Comment by sorja** [comment]
> @sorja why was ...
> merged here? you're forcing double review

Mistake, not sure what is the best way to open linear PRs

File: src/server/repository/assessmentCycle/countryActivityLog/_common/acceptedMessages.ts (hunk 5 above)
Comment (sorja) [reply src/server/repository/assessmentCycle/countryActivityLog/_common/acceptedMessages.ts:6]: I wonder if we need a migration step to recreate the activity log for countries?

Comment (minotogna) [reply src/server/repository/assessmentCycle/countryActivityLog/_common/acceptedMessages.ts:6]: sure. add a TODO in the list.

**Comment by minotogna** [comment]
> > @sorja why was ...
> > merged here? you're forcing double review
> 
> Mistake, not sure what is the best way to open linear PRs

what do you mean linear pr? we have a main branch for this!

**Review by minotogna — APPROVED** [review]

**Pull request merged**